src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -1081,10 +1082,37 @@ private static IEnumerable<Diagnostic> GetEffectiveDiagnosticsImpl(ImmutableArra
 
             var analyzerManager = new AnalyzerManager(analyzer);
 
-            Action<Exception, DiagnosticAnalyzer, Diagnostic, CancellationToken>? wrappedOnAnalyzerException =
-                onAnalyzerException == null ? null : (ex, analyzer, diagnostic, _) => onAnalyzerException(ex, analyzer, diagnostic);
-            var analyzerExecutor = AnalyzerExecutor.CreateForSupportedDiagnostics(wrappedOnAnalyzerException, analyzerManager);
-            return AnalyzerDriver.IsDiagnosticAnalyzerSuppressed(analyzer, options, analyzerManager, analyzerExecutor, analysisScope: null, severityFilter: SeverityFilter.None, CancellationToken.None);
+            Action<Exception, DiagnosticAnalyzer, Diagnostic, CancellationToken> wrappedOnAnalyzerException =
+                (ex, analyzer, diagnostic, _) => onAnalyzerException?.Invoke(ex, analyzer, diagnostic);
+
+            Func<DiagnosticAnalyzer, ImmutableArray<DiagnosticDescriptor>> getSupportedDiagnosticDescriptors = analyzer =>
+            {
+                try
+                {
+                    return analyzer.SupportedDiagnostics;
+                }
+                catch (Exception ex)
+                {
+                    AnalyzerExecutor.HandleAnalyzerException(ex, analyzer, info: null, wrappedOnAnalyzerException, analyzerExceptionFilter: null, CancellationToken.None);
+                    return ImmutableArray<DiagnosticDescriptor>.Empty;
+                }
+            };
+
+            Func<DiagnosticSuppressor, ImmutableArray<SuppressionDescriptor>> getSupportedSuppressionDescriptors = suppressor =>
+            {
+                try
+                {
+                    return suppressor.SupportedSuppressions;
+                }
+                catch (Exception ex)
+                {
+                    AnalyzerExecutor.HandleAnalyzerException(ex, analyzer, info: null, wrappedOnAnalyzerException, analyzerExceptionFilter: null, CancellationToken.None);
+                    return ImmutableArray<SuppressionDescriptor>.Empty;
+                }
+            };
+
+            return AnalyzerManager.IsDiagnosticAnalyzerSuppressed(analyzer, options, IsCompilerAnalyzer, severityFilter: SeverityFilter.None,
```

Comment:
Retained the existing functionality of this deprecated public API as per the public API review feedback: https://github.com/dotnet/roslyn/issues/67592#issuecomment-1507646644

Reply:
@sharwell - My bad `IsCompilerAnalyzer` is being accessed here, so needs to be made internal.